### PR TITLE
Israel.6027.admins and moderators page

### PR DESCRIPTION
**Pull request by Israellund**
## Link to Issue
Closes: #6027 

## Description of Changes
-consolidated logic of `ManageRoles` and `UpgradeROlesForm` into new `AdminsAndModerators` component.
-Created new page layout based on designs in Figma

## Test Plan
-Fire up my branch
-go to a community where you are Admin and click Admins & Moderators in Admin Capabilities
-check the new design against the Figma file [here](https://www.figma.com/file/ibuywR0bBcy3ZZhPXbjEJw/Community-Admin?node-id=278%3A7384&mode=dev)

## Other Considerations

https://github.com/hicommonwealth/commonwealth/assets/69872984/2825d3b7-2427-4dd0-9940-d86085a80ecd

### CODE REVIEW

File: packages/commonwealth/client/scripts/views/pages/CommunityManagement/AdminsAndModerators/AdminsAndModerators.tsx [hunk 1]
```diff
@@ -1,7 +1,174 @@
...
+
+        <UpgradeRolesForm
+          roleData={roleData}
+          onRoleUpdate={handleRoleUpdate}
+          searchTerm={searchTerm}
+          setSearchTerm={setSearchTerm}
+        />
+      </section>
+
+      <section className="right-section">
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/CommunityManagement/AdminsAndModerators/AdminsAndModerators.tsx:162]: there is a big gap b/w these sections, I suggest we make the elements inside `.left-section` to have `width: 100%`

image

File: packages/commonwealth/client/scripts/views/pages/CommunityManagement/AdminsAndModerators/AdminsAndModerators.tsx [hunk 2]
```diff
@@ -1,7 +1,174 @@
...
+        <CWText type="caption" className={ComponentType.Label}>
+          Members
+        </CWText>
+
+        <UpgradeRolesForm
+          roleData={roleData}
+          onRoleUpdate={handleRoleUpdate}
+          searchTerm={searchTerm}
+          setSearchTerm={setSearchTerm}
+        />
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/CommunityManagement/AdminsAndModerators/AdminsAndModerators.tsx:159]: We should change the checkbox and buttons, and use new designs 
image

File: packages/commonwealth/client/styles/pages/search/search_bar.scss [hunk 3]
```diff
@@ -16,7 +16,7 @@
       @include b1;
 
       border: 1px solid $neutral-300;
-      border-radius: $border-radius-rounded-corners;
+      border-radius: 6px;
```
Comment (mzparacha) [inline packages/commonwealth/client/styles/pages/search/search_bar.scss:19]: not sure if we really need this?

File: packages/commonwealth/client/styles/pages/manage_community/manage_roles.scss [hunk 4]
```diff
@@ -21,6 +21,8 @@
       height: 24px;
       justify-content: space-between;
       width: auto;
+      padding-left: 4px;
+      padding-top: 4px;
```
Comment (mzparacha) [inline packages/commonwealth/client/styles/pages/manage_community/manage_roles.scss:25]: these will effect existing community management page. Lets make sure that new pages don't effect old UI.

File: packages/commonwealth/client/scripts/views/pages/CommunityManagement/AdminsAndModerators/AdminsAndModerators.tsx [hunk 5]
```diff
@@ -1,7 +1,174 @@
...
+        },
+      });
+      const roles = res.data.result || [];
+      roles.forEach((role) => {
+        if (role.permission === AccessLevel.Admin) {
+          memberAdmins.push(role);
+        } else if (role.permission === AccessLevel.Moderator) {
+          memberMods.push(role);
+        }
+      });
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/CommunityManagement/AdminsAndModerators/AdminsAndModerators.tsx:44]: this request is also used in 3 other places, let's add a react query fetch (useQuery) for this.

File: packages/commonwealth/client/scripts/views/pages/CommunityManagement/AdminsAndModerators/AdminsAndModerators.tsx [hunk 6]
```diff
@@ -1,7 +1,174 @@
...
+  APIOrderDirection,
+} from '../../../../../scripts/helpers/constants';
+import useSearchProfilesQuery from '../../../../../scripts/state/api/profiles/searchProfiles';
+import RoleInfo from '../../../../models/RoleInfo';
+import { ComponentType } from '../../../components/component_kit/types';
+import { UpgradeRolesForm } from '../../../pages/manage_community/upgrade_roles_form';
+import { ManageRoles } from '../../manage_community/manage_roles';
+import './AdminsAndModerators.scss';
 
 const AdminsAndModerators = () => {
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/CommunityManagement/AdminsAndModerators/AdminsAndModerators.tsx:19]: we should also check 
1. is user logged in
2. is user a community or a site admin
and only then render this page.

File: packages/commonwealth/client/scripts/views/pages/CommunityManagement/AdminsAndModerators/AdminsAndModerators.tsx [hunk 7]
```diff
@@ -1,7 +1,174 @@
...
+      return {
+        ...(profile.roles[0] || {}),
+        Address: profile.addresses[0],
+        id: profile.addresses[0].id,
+        displayName: profile.profile_name || 'Anonymous',
+      };
+    });
+  }, [searchResults]);
+
+  const handleRoleUpdate = (oldRole, newRole) => {
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/CommunityManagement/AdminsAndModerators/AdminsAndModerators.tsx:83]: we have some similar logic in `ManageCommunityPage`, lets try to add a common util/custom-hook (maybe name it `useCommunityRoles`) and  and use that in both places

**Review by mzparacha — COMMENTED** [review]
Thank you for the hard work on this. Left some suggestions above.

File: packages/commonwealth/client/styles/pages/search/search_bar.scss (hunk 3 above)
Comment (Israellund) [reply packages/commonwealth/client/styles/pages/search/search_bar.scss:19]: @mzparacha If we don't add a border-radius of 6px then the search bar had 90 degree right angles, which isn't in the Figma designs. I used the same border-radius as the Search Common search bar.

File: packages/commonwealth/client/scripts/views/pages/CommunityManagement/AdminsAndModerators/AdminsAndModerators.scss [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+  @include mediumSmallInclusive {
+    grid-template-columns: 1fr;
+  }
+
+  .left-section {
+    display: flex;
+    flex-direction: column;
+    align-items: flex-start;
+    // gap: 32px;
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/pages/CommunityManagement/AdminsAndModerators/AdminsAndModerators.scss:18]: remove

File: packages/commonwealth/client/scripts/views/pages/CommunityManagement/AdminsAndModerators/AdminsAndModerators.scss [hunk 9]
```diff
@@ -0,0 +1,35 @@
+@import '../../../../../styles/shared';
+
+.AdminsAndModerators {
+  display: grid;
+  grid-template-columns: 3fr 1fr;
+  gap: 69px;
+  padding: 22px 60px;
+  padding-bottom: 100px;
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/pages/CommunityManagement/AdminsAndModerators/AdminsAndModerators.scss:8]: let's make sure that implemenation for the layout with sidebar is the same for all admin pages

File: packages/commonwealth/client/scripts/views/pages/CommunityManagement/AdminsAndModerators/AdminsAndModerators.tsx [hunk 10]
```diff
@@ -1,7 +1,174 @@
...
+    }
+
+    setAdmins(memberAdmins);
+    setMods(memberMods);
+  };
+
+  useEffect(() => {
+    if (!app.activeChainId()) {
+      return;
+    }
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/pages/CommunityManagement/AdminsAndModerators/AdminsAndModerators.tsx:56]: if we move it to react-query as @mzparacha  suggested we can remove this by using `enabled` param

**Comment by CowMuon** [comment]
Need reviews from @Miaplacidus @mzparacha @lzach83 (2 of 3)

**Comment by mzparacha** [comment]
Reviewing

**Comment by CowMuon** [comment]
Need reviews from @mzparacha @miaplacidus @zachlee83

**Comment by mzparacha** [comment]
I think somethings got missing in the merge, there was a `CommunityManagementLayout` with some other changes. 

cc: @Israellund

**Review by lzach83 — CHANGES_REQUESTED** [review]
I don't have anything additional to add here outside of what @mzparacha and @masvelio have already pointed out.

**Comment by Israellund** [comment]
I am closing this due to too many merge conflicts. I had originally created a new branch off of Malik's branch because I needed changes he had made that weren't on master yet, but while working on my branch he made extensive changes that created the merge conflicts. Because in the meantime Malik made a newly created design system for the Admin Capabilities, and because this project was under a deadline, I chose to create a new branch off of master and make changes rather than risk deleting anything Malik had done by merging.

**Pull request closed without merging**